prov/util/src/util_mem_hooks.c

```diff
@@ -567,4 +568,16 @@ void ofi_memhooks_stop(void)
 {
 }
 
-#endif
+#endif /* memhook support checks */
+
+void ofi_memhooks_init(void)
+{
+	pthread_mutex_init(&memhooks_monitor->lock, NULL);
+	dlist_init(&memhooks_monitor->list);
+}
+
+void ofi_memhooks_cleanup(void)
+{
+	assert(dlist_empty(&memhooks_monitor->list));
+	pthread_mutex_destroy(&memhooks_monitor->lock);
+}
```

Comment:
What if we just get rid of init and cleanup and require all memory monitors to have the lock and list statically initialized?